### DISCUSSION TITLE
ragna-aws

### DISCUSSION THREAD

**Discussion by NetDevAutomate** [Quansight/ragna#245] [Q&A]
The ranga-aws (https://github.com/pmeier/ragna-aws) is awesome... 

2 questions if I may:

- How do I configure Ragna to use an existing S3 Bucket as the data source i.e. where the PDF docs are stored. To clarify, the documents are already in S3 to use vs using S3 to upload to
- Are there any plans to implement any interface to a Bedrock Agent that already exists (or create a new one)?

**Comment by pmeier** [comment]
> How do I configure Ragna to use an existing S3 Bucket as the data source i.e. where the PDF docs are stored. To clarify, the documents are already in S3 to use vs using S3 to upload to

This is currently not possible. When we upload through the API, we first create an ID for each document:

https://github.com/Quansight/ragna/blob/7b176528c7524edc8bcf8754695caa5a363a8e23/ragna/deploy/_api/core.py#L121
https://github.com/Quansight/ragna/blob/7b176528c7524edc8bcf8754695caa5a363a8e23/ragna/deploy/_api/schemas.py#L18-L20

For S3 (but for local as well) we use this [ID as filename](https://github.com/pmeier/ragna-aws/blob/96059ae0f130db0719d83550268d94150e483a99/ragna_aws/_s3.py#L32) that we later [read from](https://github.com/pmeier/ragna-aws/blob/96059ae0f130db0719d83550268d94150e483a99/ragna_aws/_s3.py#L45).

Plus, we are also introducing the document into our state database:

https://github.com/Quansight/ragna/blob/7b176528c7524edc8bcf8754695caa5a363a8e23/ragna/deploy/_api/core.py#L125-L127

So you would need to replicate all of that from a script. Not impossible, but no trivial either. There is growing demand for production use case that would handle this use case. Will open an issue about this soon.

> Are there any plans to implement any interface to a Bedrock Agent that already exists (or create a new one)?

Could you elaborate on this? `ragna-aws` currently supports the foundational models. Are you talking about custom finetuned models or actual Bedrock agents? I haven't played with either so far so I can't tell how well they integrate. But from a superficial view of the docs, it should be fairly straight forward to support them.

**Reply by NetDevAutomate** [reply]
Integration with Bedrock agents, the idea is that documents would live in a S3 Bucket, the Bedrock Knowledgeable than syncs all these docs and an agent will the interact with the knowledgeable - it works really well, 

I hope to get time this week to go through https://catalog.us-east-1.prod.workshops.aws/workshops/a4bdb007-5600-4368-81c5-ff5b4154f518/en-US/90-agents/91-diy-functions to get up to speed as the workshop has a section on agents

**Reply by pmeier** [reply]
This sounds like you don't need Ragna to do that? I'm a little unsure where Ragna comes into the picture TBH. If you finished the workshop, feel free to send a PR to have a better base for discussion. The PR doesn't need to be production ready, but rather contain a proof-of-concept of what you want to achieve.
